Change Carpenter, Pot and Crate location names/variables from GF to TH

BootsinSoots opened the pull request:
## What is this fixing or adding?
Match Thieves hideout location names to Ship's counterparts

## How was this tested?
Can't yet

## If this makes graphical changes, please attach screenshots.

BootsinSoots committed:
Change Carpenter, Pot and Crate location names/variables from GF to TH

Makes the location names/variable match Ship's documentations for the Thieves' Hideout

BootsinSoots committed:
Add Rescue Carpenters region needed for union of Rescue All Carpenters

BootsinSoots committed:
Match TH crate location names to Ship logic

BootsinSoots committed:
Forgot to update the strings too

BootsinSoots committed:
Correct the ramaining Thieves Hideout locations to match Ship

aMannus commented on worlds/oot_soh/Enums.py:
```diff
@@ -92,6 +92,7 @@ class Regions(str, Enum):
     THIEVES_HIDEOUT_KITCHEN_BOTTOM = "Thieves Hideout Kitchen Bottom"
     THIEVES_HIDEOUT_KITCHEN_TOP = "Thieves Hideout Kitchen Top"
     THIEVES_HIDEOUT_BREAK_ROOM = "Thieves Hideout Break Room"
+    THIEVES_HIDEOUT_RESCUE_CARPENTERS = "Thieves Hideout Rescue Carpenters"
```
What is this one for?

BootsinSoots replied:
Have to make a region to contain the Gerudo Membership Card location. Just made another one for the Kitchen Pots, which can be accessed from 2 regions

aMannus replied:
Ah gotcha, wasn't really clear to me from this PR alone, but that's fair 👍

BootsinSoots committed:
Have to create new region for the Kitchen pots

Kitchen pots can be reached from two separate regions, have to make a union region
